### PR TITLE
server: use IsAlive() more

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind ci

/kind cleanup

> /kind dependency-change
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake
> /kind other

#### What this PR does / why we need it:
this replaces unneeded State() and UpdateContainerResources() calls,
which both saves on lock contention, as well as possibly saves on a call to `runc state`

Signed-off-by: Peter Hunt <pehunt@redhat.com>

#### Special notes for your reviewer:
replaces https://github.com/cri-o/cri-o/pull/3864
#### Does this PR introduce a user-facing change?

```release-note
none
```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
LGTM

**Review by giuseppe — APPROVED** [review]
LGTM

/retest

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests

**Comment by haircommander** [comment]
> LGTM
> assuming happy tests

happy tests? you ask for too much

**Comment by mrunalp** [comment]
Do these changes play well with kata?

**Comment by haircommander** [comment]
I believe they should, as we still set a InitPid for the container, even for shim v2. @fidencio can you ptal?

**Comment by fidencio** [comment]
Added to my list, will test it soon, please, don't merge it before I get back to this one.

**Comment by fidencio** [comment]
I know the patch needs rebase but I finally got an environment where I could test it as it is.
The basic CI on kata passed without any issues.  Knowing this, we're good to go after the rebase is done.

And sorry for the delay here.

**Pull request merged**
